<launch>


    <!-- dataset name -->
    <!-- dataset-room1_512_16, dataset-room2_512_16, dataset-corridor1_512_16 -->
    <arg name="dataset" value="dataset-room1_512_16" />
    <arg name="folder"  value="/media/patrick/RPNG FLASH 3/tum_vi" />


    <!-- MASTER NODE! -->
    <node name="$(anon estimate_vicon2gt)" pkg="vicon2gt" type="estimate_vicon2gt" output="screen" clear_params="true" required="true">
<!--    <node name="estimate_vicon2gt" pkg="vicon2gt" type="estimate_vicon2gt" output="screen" clear_params="true" required="true" launch-prefix="gdb -ex run &#45;&#45;args">-->

        <!-- bag topics -->
        <param name="topic_imu"      type="string" value="/imu0" />
        <param name="topic_vicon"    type="string" value="/vrpn_client/raw_transform" />

        <!-- bag parameters -->
        <param name="path_bag"    type="string" value="$(arg folder)/$(arg dataset).bag" />
        <param name="bag_start"   type="int"    value="0" />
        <param name="bag_durr"    type="int"    value="-1" />

        <!-- save information -->
        <param name="state_freq"         type="int"    value="100" />
        <param name="save_to_file"       type="bool"   value="true" />
        <param name="stats_path_states"  type="string" value="$(arg folder)/vicon2gt/$(arg dataset)_vicon2gt_states.csv" />
        <param name="stats_path_info"    type="string" value="$(arg folder)/vicon2gt/$(arg dataset)_vicon2gt_info.txt" />


        <!-- world parameters -->
        <rosparam param="R_BtoI">[1, 0, 0, 0, 1, 0, 0, 0, 1]</rosparam>
        <rosparam param="p_BinI">[0, 0, 0]</rosparam>
        <rosparam param="R_GtoV">[1, 0, 0, 0, 1, 0, 0, 0, 1]</rosparam>
        <param name="gravity_magnitude"          type="double" value="9.81" />
        <param name="toff_imu_to_vicon"          type="double" value="0.0" />
        <param name="num_loop_relin"             type="int"    value="0" />
        <param name="estimate_toff_vicon_to_imu" type="bool"   value="true" />
        <param name="estimate_ori_vicon_to_imu"  type="bool"   value="true" />
        <param name="estimate_pos_vicon_to_imu"  type="bool"   value="true" />


        <!-- vicon sigmas, only used if we don't get odometry -->
        <!-- sigmas: (rx,ry,rz,px,py,pz) -->
        <rosparam param="vicon_sigmas">[1e-3,1e-3,1e-3,1e-2,1e-2,1e-2]</rosparam>
        <param name="freq_pub_raw_vicon" type="double" value="1.0" />
        <param name="use_manual_sigmas"  type="bool"   value="true" />

        <!-- vi-sensor -->
        <param name="gyroscope_noise_density"      type="double"   value="1.6968e-04" />
        <param name="gyroscope_random_walk"        type="double"   value="1.9393e-05" />
        <param name="accelerometer_noise_density"  type="double"   value="2.0000e-3" />
        <param name="accelerometer_random_walk"    type="double"   value="3.0000e-3" />


    </node>


</launch>